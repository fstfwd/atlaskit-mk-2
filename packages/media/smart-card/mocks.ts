import * as fm from 'fetch-mock';

const context = '';
const googleDefinitionId = 'google';
const trelloDefinitionId = 'trello';

const serviceAuth = {
  key: 'default',
  displayName: 'Google',
  url:
    'https://outbound-auth-flow.ap-southeast-2.dev.atl-paas.net/start?containerId=f4d9cdf9-9977-4c40-a4d2-968a4986ade0&serviceKey=default',
};

const generator = {
  name: 'Google Drive',
  icon: {
    url:
      'https://ssl.gstatic.com/docs/doclist/images/infinite_arrow_favicon_5.ico',
  },
};

const resolvedBody = (definitionId: string, name: string) => ({
  meta: {
    visibility: 'restricted',
    access: 'granted',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
    name,
  },
});

const notFoundBody = (definitionId: string) => ({
  meta: {
    visibility: 'not_found',
    access: 'forbidden',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
  },
});

const unauthorisedBody = (definitionId: string) => ({
  meta: {
    visibility: 'restricted',
    access: 'unauthorized',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
  },
});

const forbiddenBody = (definitionId: string) => ({
  meta: {
    visibility: 'restricted',
    access: 'forbidden',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
  },
});

const delayP = (n: number) => new Promise(res => setTimeout(res, n));

export const mockMultipleCards = () => {
  let c1 = 0;
  let c2 = 0;
  fm.mock('*', async (_, opts: any) => {
    await delayP(1000);

    const resourceUrl = JSON.parse(opts.body).resourceUrl;

    if (resourceUrl.startsWith('google')) {
      c1++;
      const step = Math.floor((c1 - 1) / 5);
      if (step === 0) {
        return;
      }
      if (step === 1) {
        return unauthorisedBody(googleDefinitionId);
      }
      if (step === 2) {
        return;
      }
      if (step === 3) {
        return forbiddenBody(googleDefinitionId);
      }
      return resolvedBody(googleDefinitionId, resourceUrl);
    }

    if (resourceUrl.startsWith('trello')) {
      switch (++c2) {
        case 1:
          return unauthorisedBody(trelloDefinitionId);
        case 2:
          return forbiddenBody(trelloDefinitionId);
        default:
          return resolvedBody(trelloDefinitionId, resourceUrl);
      }
    }

    throw new Error('Unkonws request type');
  });
};

export const mockSingleCardWorkflow = () => {
  let c = 0;
  fm.mock(
    'https://api-private.stg.atlassian.com/object-resolver/resolve',
    () => {
      return delayP(10000).then(() => {
        switch (++c) {
          case 1:
            return unauthorisedBody;
          case 2:
            return forbiddenBody;
          default:
            return resolvedBody;
        }
      });
    },
  );
};
