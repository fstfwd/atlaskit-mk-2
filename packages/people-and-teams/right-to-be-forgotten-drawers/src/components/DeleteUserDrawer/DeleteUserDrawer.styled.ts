import styled from 'styled-components';

export const DrawerInner = styled.div`
  display: flex;
  justify-content: center;
`;

export const ContentFooter = styled.div`
  display: flex;
  justify-content: flex-end;
`;

/** Delete me once real content is present in the drawer */
export const PlaceholderContent = styled.div`
  width: 352px;
`;
