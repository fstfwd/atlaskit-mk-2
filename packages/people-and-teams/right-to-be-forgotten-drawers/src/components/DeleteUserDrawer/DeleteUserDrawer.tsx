import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import Drawer from '@atlaskit/drawer';
import CrossIcon from '@atlaskit/icon/glyph/cross';
import Button from '@atlaskit/button';

import { Screens } from '../../constants';
import { User } from '../../types';
import OverviewScreen from '../OverviewScreen';
import ContentPreviewScreen from '../ContentPreviewScreen';
import * as Styled from './styled';
import Footer from '../Footer';
import { commonMessages } from '../../messages';

interface Props {
  isOpen: boolean;
  deleteAccount: () => void;
  onClose: () => void;
  user: User;
  currentUserId: string;
}

export class DeleteUserDrawer extends React.Component<Props> {
  state = {
    currentScreen: Screens.OVERVIEW,
  };

  screens = [Screens.OVERVIEW, Screens.CONTENT_PREVIEW];

  nextScreen = () => {
    const currentScreenIdx = this.screens.indexOf(this.state.currentScreen);
    const nextScreen =
      currentScreenIdx < this.screens.length - 1
        ? currentScreenIdx + 1
        : this.screens.length - 1;
    this.setState({ currentScreen: this.screens[nextScreen] });
  };

  previousScreen = () => {
    const currentScreenIdx = this.screens.indexOf(this.state.currentScreen);
    const previousScreen = currentScreenIdx - 1 >= 0 ? currentScreenIdx - 1 : 0;
    this.setState({ currentScreen: this.screens[previousScreen] });
  };

  render() {
    const { currentUserId, deleteAccount, isOpen, onClose, user } = this.props;
    return (
      <Drawer
        icon={props => <CrossIcon label="" {...props} size="medium" />}
        isOpen={isOpen}
        onClose={onClose}
        width="full"
      >
        <Styled.DrawerInner>
          {this.state.currentScreen === Screens.OVERVIEW && (
            <OverviewScreen
              isCurrentUser={user.id === currentUserId}
              onCancel={this.props.onClose}
              onNext={this.nextScreen}
              user={user}
            />
          )}
          {this.state.currentScreen === Screens.CONTENT_PREVIEW && (
            <ContentPreviewScreen
              onPrevious={this.previousScreen}
              onDeleteAccount={deleteAccount}
              user={user}
            />
          )}
          <Footer
            numScreens={this.screens.length}
            currentScreenIdx={this.screens.indexOf(this.state.currentScreen)}
            onCancel={onClose}
            onNext={this.nextScreen}
            onPrevious={this.previousScreen}
            submitButton={
              <Button appearance="primary" onClick={deleteAccount}>
                <FormattedMessage {...commonMessages.deleteAccount} />
              </Button>
            }
          />
        </Styled.DrawerInner>
      </Drawer>
    );
  }
}

export default DeleteUserDrawer;
