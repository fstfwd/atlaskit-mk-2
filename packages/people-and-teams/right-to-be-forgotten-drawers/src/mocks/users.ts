export const catherineHirons = {
  id: 'chirons',
  fullName: 'Catherine Hirons',
  email: 'catherine.hirons@acme.com',
};
