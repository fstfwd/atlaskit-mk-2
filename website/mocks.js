// this file is a placeholder file for the generated HTTP mocks

import modules from './src/assets/**/*';

console.log(modules);
console.log('MOCKING HTTP REQUESTS!!!');
