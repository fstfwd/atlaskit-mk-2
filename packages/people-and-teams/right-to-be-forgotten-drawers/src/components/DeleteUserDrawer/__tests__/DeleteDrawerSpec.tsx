import * as React from 'react';
import { shallow } from 'enzyme';

import { catherineHirons } from '../../../mocks/users';
import DeleteUserDrawer from '../DeleteUserDrawer';

const defaultProps = {
  isOpen: false,
  deleteAccount: jest.fn(),
  onClose: jest.fn(),
  user: catherineHirons,
  currentUserId: catherineHirons.id,
};

const render = (props = {}) =>
  shallow(<DeleteUserDrawer {...defaultProps} {...props} />);

test('isOpen snapshot', () => {
  expect(render()).toMatchSnapshot();
});

describe('nextScreen()', () => {
  test('Goes to next screen', () => {
    const wrapper = render();
    const screens = wrapper.instance().screens;
    wrapper.instance().nextScreen();
    wrapper.update();
    expect(
      screens.findIndex(s => s.id === wrapper.state().currentScreenId),
    ).toBe(1);
  });

  test('No-op if on last screen', () => {
    const wrapper = render();
    const screens = wrapper.instance().screens;
    const lastScreenIdx = screens.length - 1;
    wrapper.setState({ currentScreenId: screens[lastScreenIdx].id });
    wrapper.instance().nextScreen();
    wrapper.update();
    expect(
      screens.findIndex(s => s.id === wrapper.state().currentScreenId),
    ).toBe(lastScreenIdx);
  });
});

describe('previousScreen()', () => {
  test('Goes to previous screen', () => {
    const wrapper = render();
    const screens = wrapper.instance().screens;
    const lastScreenIdx = screens.length - 1;
    wrapper.setState({ currentScreenId: screens[lastScreenIdx].id });
    wrapper.instance().previousScreen();
    wrapper.update();
    expect(
      screens.findIndex(s => s.id === wrapper.state().currentScreenId),
    ).toBe(lastScreenIdx - 1);
  });

  test('Goes to next screen', () => {
    const wrapper = render();
    const screens = wrapper.instance().screens;
    wrapper.instance().previousScreen();
    wrapper.update();
    expect(
      screens.findIndex(s => s.id === wrapper.state().currentScreenId),
    ).toBe(0);
  });
});
