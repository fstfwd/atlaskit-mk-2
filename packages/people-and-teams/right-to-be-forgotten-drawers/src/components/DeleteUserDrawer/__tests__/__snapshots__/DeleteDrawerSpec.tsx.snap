// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isOpen snapshot 1`] = `
<Drawer
  icon={[Function]}
  isOpen={false}
  onClose={[Function]}
  width="full"
>
  <styled.div>
    <Component
      isCurrentUser={true}
      onCancel={[Function]}
      onNext={[Function]}
      user={
        Object {
          "email": "catherine.hirons@acme.com",
          "fullName": "Catherine Hirons",
          "id": "chirons",
        }
      }
    />
    <Footer
      currentScreenIdx={0}
      numScreens={2}
      onCancel={[Function]}
      onNext={[Function]}
      onPrevious={[Function]}
      submitButton={
        <Button
          appearance="primary"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Delete account"
            description="Text that appears in a button, link or heading for the action of deleting an account"
            id="right-to-be-forgotten.delete-account"
            values={Object {}}
          />
        </Button>
      }
    />
  </styled.div>
</Drawer>
`;
